spec/models/api/trainee_filter_params_spec.rb

```diff
@@ -4,43 +4,53 @@
 
 describe Api::TraineeFilterParams do
   describe "since validation" do
-    describe "since data validation" do
+    context "when in ISO 8601" do
       it "correctly formatted dates are valid" do
-        expect(described_class.new(since: "2020-01-01").valid?).to be(true)
+        expect(described_class.new(since: "2024-03-15T12:15:37.879Z").valid?).to be(true)
       end
+    end
+
+    context "when not in ISO 8601" do
+      it "is not valid" do
+        expect(described_class.new(since: "2020-01-01").valid?).to be(false)
+      end
+    end
 
+    context "when non dates" do
       it "non-dates are not valid and return an error message" do
-        filter_params = described_class.new(since: "2020-01-32")
-        expect(filter_params.valid?).to be(false)
-        expect(filter_params.errors[:since]).to include("2020-01-32 is not a valid date")
+        %w[2020-01-32 123 12345 date].each do |since|
```

Comment:
So, I think we might be getting into the details of date standards here! Depending on what version of ISO8601 is being used, `12345` can be a [valid (truncated) format](https://en.wikipedia.org/wiki/ISO_8601#Truncated_representations_(deprecated)). It will be interpreted as `YYDDD` so in this case `YY = 12 = 2012` and `DDD = 345 = 345th day of the year = 10 December`

Reply:
So, it looks like our Ruby version still supports those truncated date formats and that test is behaving as expected.